contracts/MetawoodAuction.sol

```diff
@@ -0,0 +1,330 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+import "./interfaces/IMetawoodNFT.sol";
+import "@openzeppelin/contracts/utils/Counters.sol";
+import "@openzeppelin/contracts/utils/math/SafeMath.sol";
+import "@openzeppelin/contracts/token/ERC1155/utils/ERC1155Holder.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+
+contract MetawoodAuction is ERC1155Holder, Ownable, ReentrancyGuard {
+    using SafeMath for uint256;
+    using Counters for Counters.Counter;
+
+    Counters.Counter private _auctionCounter;
+    IMetawoodNFT public metawoodNFT;
+
+    uint256 public bidIncreaseThreshold = 200;
+
+    constructor(address _nftContract) {
+        metawoodNFT = IMetawoodNFT(_nftContract);
+    }
+
+    enum AuctionState {
+        CLOSED,
+        ACTIVE
+    }
+
+    //single struct for handling bid, tbd?
+    struct Auction {
+        uint256 id;
+        IMetawoodNFT nftContract;
+        uint256 nftId; // NFT Id
+        address payable creator; // Creator of the Auction
+        uint256 bidIncreaseThreshold;
```

Comment:
This has already been defined so can be removed

Reply:
This is kept if the auction creator wants to specify by which amount the bid should increase, :thinking:  the user itself should decide how much he wants the bid to increase instead of the platform constant, we can discuss this further